### ISSUE TITLE
Stopped working in Google Docs.

### ISSUE THREAD

**Issue by braylaoshi** [cschiller/zhongwen#61]
I've been using Zhongwen for years and it has been a great help with my work. I use it mostly within Google docs for my work. Today it suddenly stopped working on Google Docs even though it still works on other sites. Help!

**Comment by cschiller** [comment]
I just tested it, it still works for me. Try these steps:

- Reload the page in the browser.
- Turn Zhongwen off and then back on by clicking on the icon.

If it still doesn't work, try reinstalling it.

Hope that helps!

**Comment by braylaoshi** [comment]
Thank for the response. It’s still not working, but none of the other pop-ups are either, so it must be a problem with chrome or my laptop, but I wouldn’t even know where to start with that. Anyway, hope I can get it working again at some point. Thanks!

Respectfully Yours,
Aaron Bray (高山)

**Comment by ienablemuch** [comment]
> I've been using Zhongwen for years and it has been a great help with my work. I use it mostly within Google docs for my work. Today it suddenly stopped working on Google Docs even though it still works on other sites. Help!

https://workspaceupdates.googleblog.com/2021/05/Google-Docs-Canvas-Based-Rendering-Update.html

**Comment by cschiller** [comment]
Yes, canvas-based rendering in Google Docs might in fact break the Zhongwen Chrome extension. Unfortunately, I don't see a workaround for this.

**Comment by jensuzzane** [comment]
kind of awkward workaround but it still works in a draft email in gmail. i just copied and pasted the text i was reading from google docs into an email.

**Comment by jshmrsn** [comment]
I found a partial solution! In Google docs, if you select a chunk of text, then right click on it, one of the menu items will be "Define 'text you selected'". Don't actually click on this menu item, instead just hover over the 'text you selected', and Zhongwen will pop up as normal. Slightly less convenient, but basically just one more click away. Hopefully this functionality lasts.

**Comment by chinese-words-separator** [comment]
> I found a partial solution! In Google docs, if you select a chunk of text, then right click on it, one of the menu items will be "Define 'text you selected'". Don't actually click on this menu item, instead just hover over the 'text you selected', and Zhongwen will pop up as normal. Slightly less convenient, but basically just one more click away. Hopefully this functionality lasts.

That is a good work-around. If you need longer text to proofread, or if the `Define 'text you selected'` functionality is changed to canvas tech too, following approach is another work-around, [CWS](https://chrome.google.com/webstore/detail/chinese-words-separator/gacfacdpfimbkgcnlegknnmcccjgcbnp) has an HTML-based Note tool that can be overlaid on top of Google Docs:

![work-around-for-google-doc-move-to-canvas-tech](https://user-images.githubusercontent.com/91402471/193962600-2a5f8c9e-ecf9-4a2b-a66b-65063099fac5.gif)

Zhongwen does not work on CWS's Note tool though due to CWS's Note being embedded in CWS's page, Chrome prevents extension to interact with another extension's page

If you need Zhongwen work with longer text, Zhongwen can use the same approach, that is, by overlaying an HTML widget on top of Google Docs
